unumpy/tests/test_numpy.py

```diff
@@ -563,3 +563,13 @@ def test_class_overriding():
         assert np.dtype("float64") == onp.float64
         assert isinstance(np.dtype("float64"), onp.dtype)
         assert issubclass(onp.ufunc, np.ufunc)
+
+    try:
+        with ua.set_backend(CupyBackend, coerce=True):
+            assert isinstance(cp.add, np.ufunc)
+            assert isinstance(cp.dtype("float64"), np.dtype)
+            assert np.dtype("float64") == cp.float64
+            assert isinstance(np.dtype("float64"), cp.dtype)
+            assert issubclass(cp.ufunc, np.ufunc)
+    except AttributeError:
```

Comment:
This does seem to be working, although I think this try-catch block is covering too much of the test. Could you try replacing it with:
```python
if 'CupyBackend' in globals():
```

Reply:
https://github.com/Quansight-Labs/unumpy/blob/70a84667381c32c1df1dd30cc3b12b61a00b5908/unumpy/tests/test_numpy.py#L41

Wouldn't it be in `globals()` either way because of the above?

Reply:
Maybe change that line to `CupyBackend = None` and check for `CupyBackend is None` instead?

Reply:
Good point, you could do `cp in globals()` or in the try block for the import just create a variable `has_cupy`.

Reply:
Not sure if CuPy has to assigned an `object()` for something to work, I would guess not. @peterbell10? If not then the `None` option seems good to me.

Reply:
There are several places where I've put `if backend is CupyBackend` so it only has to exist and be distinct from other backends.
https://github.com/Quansight-Labs/unumpy/blob/70a84667381c32c1df1dd30cc3b12b61a00b5908/unumpy/tests/test_numpy.py#L235

Reply:
Maybe `hasattr(CuPyBackend, "__ua_function__")`?